File: Android/app/src/main/java/com/dreamproject/inwords/domain/interactor/translation/TranslationSyncInteractorImpl.java
package com.dreamproject.inwords.domain.interactor.translation;

import com.dreamproject.inwords.data.sync.TranslationSyncController;

import javax.inject.Inject;

import io.reactivex.Completable;
import io.reactivex.schedulers.Schedulers;

public class TranslationSyncInteractorImpl implements TranslationSyncInteractor {
    private TranslationSyncController syncController;

    @Inject
    TranslationSyncInteractorImpl(TranslationSyncController syncController) {
        this.syncController = syncController;
    }

    @Override
    public Completable presyncOnStart() {
        return syncController.presyncOnStart()
                .subscribeOn(Schedulers.io())
                .doOnError(Throwable::printStackTrace)
                .ignoreElement();
    }

    @Override
    public Completable trySyncAllReposWithCache() {
        return syncController.trySyncAllReposWithCache()
                .subscribeOn(Schedulers.io());
    }

    @Override
    public void notifyDataChanged() {
        syncController.notifyDataChanged();
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/translation/domain/interactor/TranslationSyncInteractorImpl.kt
package ru.inwords.inwords.translation.domain.interactor

import android.util.Log
import io.reactivex.Completable
import ru.inwords.inwords.translation.data.sync.TranslationSyncController
import javax.inject.Inject

class TranslationSyncInteractorImpl @Inject
internal constructor(private val syncController: TranslationSyncController) : TranslationSyncInteractor {

    override fun presyncOnStart(): Completable {
        return syncController.presyncOnStart()
                .doOnError { Log.e(javaClass.simpleName, it.message.orEmpty()) }
                .ignoreElement()
    }

    override fun trySyncAllReposWithCache(): Completable {
        return syncController.trySyncRemoteReposWithLocal()
    }

    override fun notifyDataChanged() = syncController.notifyDataChanged()
}
